Group related items by mainstream browse page

rubenarakelyan opened the pull request:
Currently, related items are grouped based on having a common parent or grandparent with the content item they are linked to. This commit changes the grouping logic to group by common `mainstream_browse_pages`, in order to match the currently implemented grouping logic.

jackscotti commented:
👍 might want to update the changelog too

tijmenb commented:
👍 for intent, but needs a thorough read through to update the method names & comments to reflect new behaviour.

rubenarakelyan commented:
@jackscotti @tijmenb I've updated the comments and method/variable names to be more descriptive (hopefully) and also added the (unreleased) related items functionality to the changelog.
